### ISSUE TITLE
Filter empty labels

### ISSUE THREAD

**Issue by C-hongfei** [ultralytics/yolov5#12632]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

My dataset contains some background images without labels, how can I only load labeled images and not background images during training without changing the dataset？

**Comment by glenn-jocher** [comment]
@C-hongfei hello! 👋

To ensure YOLOv5 only loads images with labels during training, you can use the `--filter-empty` training argument. This will automatically skip images without corresponding label files.

Just add this flag when you start your training command, and you should be good to go!

For more details on training arguments, you can refer to our documentation.

Happy training! 😊

**Comment by C-hongfei** [comment]
> @C-hongfei hello! 👋
> 
> To ensure YOLOv5 only loads images with labels during training, you can use the `--filter-empty` training argument. This will automatically skip images without corresponding label files.
> 
> Just add this flag when you start your training command, and you should be good to go!
> 
> For more details on training arguments, you can refer to our documentation.
> 
> Happy training! 😊

Thanks for the quick answer, but I have a problem training with the --filter-empty parameters
My training command is as:
`python train.py --img 640 --batch 72 --epoch 50 --data data/mydata.yaml --cfg models/yolov5s.yaml --weights weights/yolov5s.pt --device 0,1,2,3,4,5 --filter-empty`
But it doesn't seem to have this parameter:
usage: train.py [-h] [--weights WEIGHTS] [--cfg CFG] [--data DATA] [--hyp HYP]
                [--epochs EPOCHS] [--batch-size BATCH_SIZE] [--imgsz IMGSZ]
                [--rect] [--resume [RESUME]] [--nosave] [--noval]
                [--noautoanchor] [--noplots] [--evolve [EVOLVE]]
                [--bucket BUCKET] [--cache [CACHE]] [--image-weights]
                [--device DEVICE] [--multi-scale] [--single-cls]
                [--optimizer {SGD,Adam,AdamW}] [--sync-bn] [--workers WORKERS]
                [--project PROJECT] [--name NAME] [--exist-ok] [--quad]
                [--cos-lr] [--label-smoothing LABEL_SMOOTHING]
                [--patience PATIENCE] [--freeze FREEZE [FREEZE ...]]
                [--save-period SAVE_PERIOD] [--seed SEED]
                [--local_rank LOCAL_RANK] [--entity ENTITY]
                [--upload_dataset [UPLOAD_DATASET]]
                [--bbox_interval BBOX_INTERVAL]
                [--artifact_alias ARTIFACT_ALIAS]
train.py: error: unrecognized arguments: --filter-empty

**Comment by glenn-jocher** [comment]
Apologies for the confusion, @C-hongfei. It seems there was a misunderstanding. The `--filter-empty` parameter is not available in the YOLOv5 training script.

To work around this, you could manually remove the paths to images without labels from your dataset's `.txt` files that list the training and validation images. This would ensure that only images with corresponding labels are loaded during training.

If you need further assistance on how to modify your dataset, please refer to our documentation or feel free to ask here. Keep up the great work! 🚀

**Comment by C-hongfei** [comment]
> Apologies for the confusion, @C-hongfei. It seems there was a misunderstanding. The `--filter-empty` parameter is not available in the YOLOv5 training script.
> 
> To work around this, you could manually remove the paths to images without labels from your dataset's `.txt` files that list the training and validation images. This would ensure that only images with corresponding labels are loaded during training.
> 
> If you need further assistance on how to modify your dataset, please refer to our documentation or feel free to ask here. Keep up the great work! 🚀

i will do that, thank you

**Comment by glenn-jocher** [comment]
You're welcome, @C-hongfei! If you have any more questions or need further assistance in the future, don't hesitate to reach out. Best of luck with your training! 😄👍
